lib/main.js

```diff
@@ -60,7 +67,31 @@ export default {
         // Add the file to be linted
         parameters.push(filePath);
 
-        const output = await helpers.exec(executablePath, parameters, options);
+        let output;
+        try {
+          output = await helpers.exec(executablePath, parameters, options);
+        } catch (e) {
+          showErrorNotification(e.message);
+          return null;
+        }
+
+        const messages = [];
+
+        if (output.exitCode !== 0) {
+          if (output.stderr.toLowerCase().startsWith(warning)) {
+            messages.push({
+              severity: warning,
+              excerpt: `haml-lint: ${output.stderr}`,
+              location: {
+                file: filePath,
+                // first line of the file
+                position: [[0, 0], [0, Infinity]],
+              },
+            });
+          } else {
+            showErrorNotification(output.stderr);
```

Comment:
Shouldn't this `return null;` here as well to signify to the consumer that any current messages shouldn't be updated?